src/gobapi/auth/auth_query.py

```diff
@@ -67,15 +70,22 @@ def get_suppressed_columns(self):
         """
         The suppressed columns are the columns that require a role that the user doesn't have
         """
-        return [attr for attr, auth in self.get_checked_columns().items() if not self._is_authorized_for(auth)]
+        if self.allows_access():
+            return [attr for attr, auth in self.get_checked_columns().items() if not self._is_authorized_for(auth)]
+        else:
+            return self._attributes
 
     def filter_row(self, row):
         """
         Set all columns in the row that should be suppressed to None
         """
-        suppressed_columns = self.get_suppressed_columns()
-        for column in [c for c in suppressed_columns if c in row]:
-            row[column] = None
+        if self.allows_access():
```

Comment:
Duplicate functionality. I think looping over suppressed_columns should be enough, as the check for allow_access is also done in get_suppressed_columns.